tests/hardhat/Fork/diamondTest.ts

```diff
@@ -0,0 +1,414 @@
+
+          comptrollerLens = await unitroller.comptrollerLens();
+          const comptrollerLensUpgrade = await diamondUnitroller.comptrollerLens();
+          expect(comptrollerLens).to.equal(comptrollerLensUpgrade);
+
+          // cheking all public mappings
+          market = await unitroller.markets(vBUSD.address);
+          const marketUpgrade = await diamondUnitroller.markets(vBUSD.address);
+          expect(market.collateralFactorMantissa).to.equal(marketUpgrade.collateralFactorMantissa);
+          expect(market.isListed).to.equal(marketUpgrade.isListed);
+          expect(market.isVenus).to.equal(marketUpgrade.isVenus);
+
+          venusBorrowerIndex = await unitroller.venusBorrowerIndex(vBUSD.address, busdHolder.address);
+          const venusBorrowerIndexUpgrade = await diamondUnitroller.venusBorrowerIndex(
+            vBUSD.address,
+            busdHolder.address,
+          );
+          expect(venusBorrowerIndex).to.equal(venusBorrowerIndexUpgrade);
+
+          venusSupplierIndex = await unitroller.venusSupplierIndex(vBUSD.address, busdHolder.address);
+          const venusSupplierIndexUpgrade = await diamondUnitroller.venusSupplierIndex(
+            vBUSD.address,
+            busdHolder.address,
+          );
+          expect(venusSupplierIndex).to.equal(venusSupplierIndexUpgrade);
+
+          const venusBorrowSpeeds = await unitroller.venusBorrowSpeeds(vUSDT.address);
+          const venusBorrowSpeedsUpgrade = await diamondUnitroller.venusBorrowSpeeds(vUSDT.address);
+          const venusSupplySpeeds = await unitroller.venusSupplySpeeds(vUSDT.address);
+          const venusSupplySpeedsUpgrade = await diamondUnitroller.venusSupplySpeeds(vUSDT.address);
+
+          expect(venusBorrowSpeeds).to.equal(venusBorrowSpeedsUpgrade);
+          expect(venusSupplySpeeds).to.equal(venusSupplySpeedsUpgrade);
+        });
+      });
+    });
+
+    describe("Verify states of diamond Contract", () => {
+      describe("Diamond setters", () => {
+        it("setting market supply cap", async () => {
+          const currentSupplyCap = (await diamondUnitroller.supplyCaps(vBUSD.address)).toString();
+          await diamondUnitroller.connect(owner)._setMarketSupplyCaps([vBUSD.address], [parseUnits("100000", 18)]);
+          expect(await diamondUnitroller.supplyCaps(vBUSD.address)).to.equals(parseUnits("100000", 18));
+          await diamondUnitroller
+            .connect(owner)
+            ._setMarketSupplyCaps([vBUSD.address], [parseUnits(currentSupplyCap, 0)]);
+          expect(await diamondUnitroller.supplyCaps(vBUSD.address)).to.equals(parseUnits(currentSupplyCap, 0));
+        });
+
+        it("setting close factor", async () => {
+          const currentCloseFactor = (await diamondUnitroller.closeFactorMantissa()).toString();
+          await diamondUnitroller.connect(owner)._setCloseFactor(parseUnits("10000", 18));
+          expect(await diamondUnitroller.closeFactorMantissa()).to.equals(parseUnits("10000", 18));
```

Comment:
The `closeFactorMaxMantissa` is `0.9e18`

```suggestion
          await diamondUnitroller.connect(owner)._setCloseFactor(parseUnits("0.8", 18));
          expect(await diamondUnitroller.closeFactorMantissa()).to.equals(parseUnits("0.8", 18));
```